### PR TITLE
Fixes boolean_prefix_lint showing for overriden methods

### PR DESCRIPTION
**Pull request by imsamgarg**
# Description

Handle the case when the method is a overriden method. We don't need to show lint on overriden methods.

Fixes boolean_prefix_lint showing for overriden methods

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor
- [ ] Documentation
- [ ] Other

## Checklist

- [x] I have read the [CONTRIBUTING.md][contributing_link] document.
- [x] I have performed a self-review of my code.
- [ ] I have linked the issue ticket in the description.
- [ ] I have made the necessary changes to the documentation.
- [x] I have checked the formatting with `dart format .`
- [x] I have analyzed my code with `dart analyze .`
- [x] I have run `dart run custom_lint example` to check for any custom linting issues.

[contributing_link]: https://github.com/charlescyt/pyramid_lint/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: lib/src/lints/dart/boolean_prefix.dart [hunk 1]
```diff
@@ -74,6 +74,7 @@ class BooleanPrefix extends DartLintRule {
     context.registry.addMethodDeclaration((node) {
       final returnType = node.returnType?.type;
       if (returnType == null || !returnType.isDartCoreBool) return;
+      if (node.declaredElement?.hasOverride ?? true) return;
```
Comment (charlescyt) [inline lib/src/lints/dart/boolean_prefix.dart:77]: I think this should be `if (node.declaredElement?.hasOverride ?? false)`.
I personally prefer `if (node.declaredElement?.hasOverride == true)`.
Using nullable boolean with `??` is sometimes confusing.

Comment (imsamgarg) [reply lib/src/lints/dart/boolean_prefix.dart:77]: > I think this should be `if (node.declaredElement?.hasOverride ?? false)`.

No this means if the declaredElement is null then the if block will be ignored and rest of the program will execute which we don't want if the declaredElement is null (ast is not yet resolved ).

> I personally prefer `if (node.declaredElement?.hasOverride == true)`.

This is also incorrect due to the same argument as above.

if you find it confusing we can write in this way

```dart
final elem=node.declaredElement;
if(elem == null || elem.hasOverride) return ;
```

Comment (charlescyt) [reply lib/src/lints/dart/boolean_prefix.dart:77]: I see. Let's do it this way.
But use `element` instead of `elem`.

**Review by imsamgarg — COMMENTED** [review]
Changes done. Please review it.

**Pull request merged**